tool/crawl.dart

```diff
@@ -137,9 +137,10 @@ Future<String> findSinceLinter(String lint) async {
 
 Future<int> _readLatestMinorVersion() async {
   var contents = await File('pubspec.yaml').readAsString();
-  YamlMap pubspec = loadYamlNode(contents);
+  YamlMap pubspec = loadYamlNode(contents) as YamlMap;
   // 0.1.79 or 0.1.79-dev
-  return int.parse(pubspec['version'].split('.').last.split('-').first);
+  return int.parse(
+      pubspec['version'].split('.').last.split('-').first as String);
```

Comment:
Consider moving the cast earlier: `(pubspec['version'] as String).split('.')`. It would allow stronger checking of the code following the cast.

Reply:
Good idea.  In fact, I wonder if we couldn't add a hint in cases like this...

Reply:
The problem is that `pubspec['version']` is `dynamic`, which means we can't know which `split` is being invoked. We'd basically need to work backwards by finding all declarations of `first` and look for those that return a `String`, then look for all implementations of `split` that return one of the types that implement one of the declarations of `first`, then ..., and then hope that there's only one choice of types to which to `pubspec['version']` could be cast in order to make it all type check correctly.